staging/src/k8s.io/api/resource/v1/types.go

```diff
@@ -171,19 +174,26 @@ type ResourceSliceSpec struct {
 	// SharedCounters defines a list of counter sets, each of which
 	// has a name and a list of counters available.
 	//
-	// The names of the SharedCounters must be unique in the ResourceSlice.
+	// The names of the counter sets must be unique in the ResourcePool.
+	//
+	// Only one of Devices and SharedCounters can be set in a ResourceSlice.
 	//
-	// The maximum number of counters in all sets is 32.
+	// The maximum number of counter sets is 8.
 	//
 	// +optional
 	// +listType=atomic
+	// +k8s:listType=atomic
+	// +k8s:unique=map
+	// +k8s:listMapKey=name
 	// +featureGate=DRAPartitionableDevices
+	// +zeroOrOneOf=ResourceSliceType
+	// +k8s:maxItems=8
```

Comment:
All DV fields need to be marked +k8s:optional or +k8s:required.  This also needs a:
```
// +k8s:optional
```
tag.  This should be done across all API versions